bindings/java/Cargo.toml

```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 [package]
-name = "opendal-java"
+name = "opendaljni"
```

Comment:
I prefer to maintain the name 'opendal-java' for the following reasons: 

- It helps users who are not familiar with the Java ecosystem to understand what this is.
- It maintains consistency with the naming style used for other bindings.